src/routes/admin/_admin.contacts.tsx

```diff
@@ -100,37 +102,18 @@ export function Contacts() {
   }, [tab]);
 
   useEffect(() => {
-    navigate({
-      search: (old) => {
-        const id = getContactId(filter);
-        const paymail = getContactPaymail(filter);
-        return {
-          ...old,
-          id,
-          paymail,
-          pubKey: !id && !paymail && filter ? filter : undefined,
-        };
-      },
-      replace: true,
-    });
-  }, [debouncedFilter]);
-
-  useEffect(() => {
-    setFilter(id || paymail || pubKey || '');
-    navigate({
-      search: (old) => {
-        const id = getContactId(filter);
-        const paymail = getContactPaymail(filter);
-        return {
-          ...old,
-          id,
-          paymail,
-          pubKey: !id && !paymail && filter ? filter : undefined,
-        };
-      },
-      replace: true,
-    });
-  }, [id, paymail, pubKey]);
+    if (filter === '') {
+      setID(undefined);
+      setPaymail(undefined);
+      setPubKey(undefined);
+    } else if (filter.length === CONTACT_ID_LENGTH) {
+      setID(filter);
+    } else if (filter.includes('@')) {
+      setPaymail(filter);
+    } else {
+      setPubKey(filter);
+    }
+  }, [filter]);
```

Comment:
Small optimization suggestion for early return to reduce the nesting but also Falsey check which would also include empty string? But not sure if there are scenarios that it would be required to check for that?

```suggestion
   if (!filter) {
    setID(undefined);
    setPaymail(undefined);
    setPubKey(undefined);
    return;
  }

  if (filter.length === CONTACT_ID_LENGTH) {
    setID(filter);
  } else if (filter.includes('@')) {
    setPaymail(filter);
  } else {
    setPubKey(filter);
  }
}, [filter]);
```